src/js/components/sidenav.js

```diff
@@ -92,30 +92,52 @@ export default class Sidenav extends Component {
         }
     
         targetList.hasAttribute('hidden')
-          ? this.open(toggleButton, targetList)
-          : this.close(toggleButton, targetList);
+          ? this.open(toggleId)
+          : this.close(toggleId);
       },
 
-      open(toggleButton, targetList) {
-        const openEvent = Component.dispatchCustomEvent('sidenavListOpen', toggleButton, {
-          id: toggleButton.dataset.rvtSidenavToggle
+      open(toggleId) {
+        const toggle = this.element.querySelector(
+          `[${this.toggleAttribute}="${toggleId}"]`
+        );
+        const targetList = this.element.querySelector(
+          `[${this.listAttribute}="${toggleId}"]`
+        );
+
+        if (!toggle || !targetList) {
+          console.warn(`No such toggle '${toggleId}' in Sidenav.open()`);
+          return;
+        }
+
+        const openEvent = Component.dispatchCustomEvent('sidenavListOpened', toggle, {
+          id: toggleId
         });
     
         if (!openEvent) return;
     
-        toggleButton.setAttribute('aria-expanded', 'true');
+        toggle.setAttribute('aria-expanded', 'true');
         targetList.removeAttribute('hidden');
       },
     
-      close(toggleButton, targetList) {
-        // FIXME: Changed to sidenavListClosed to match tense of other events
-        const closeEvent = Component.dispatchCustomEvent('sidenavListClose', toggleButton, {
-          id: toggleButton.dataset.rvtSidenavToggle
+      close(toggleId) {
+        const toggle = this.element.querySelector(
+          `[${this.toggleAttribute}="${toggleId}"]`
+        );
+        const targetList = this.element.querySelector(
+          `[${this.listAttribute}="${toggleId}"]`
+        );
+        if (!toggle || !targetList) {
```

Comment:
Nitpick, but maybe add an empty line before this `if` statement to match the `open` method above.

Reply:
Yeah, I didn't bother with any styling/formatting stuff in this PR. My intention is to do a style/refactoring pass in the near future on all the components.